src/linear_samplers/block_row_rand_cyclic.jl

```diff
@@ -0,0 +1,56 @@
+# This file is part of RLinearAlgebra.jl 
+"""
+    LinSysBlkRowRandCyclic <: LinSysBlkRowSampler
+
+A mutable structure with fields to handle randomly permuted block sampling. After each cycle, 
+a new random ordering is created. 
+
+# Fields
+- `n_blocks::Int64` - Variable that contains the number of blocks overall.
+- `order::Union{Vector{Int64}, Nothing}` - The order that the blocks will be used to generate updates.
+- `blocks::Union{Vector{Vector{Int64}}, Nothing}` - The list of all the row in each block.
+
+Calling `LinSysVecColBlockRandCyclic()` defaults to setting `n_blocks` to 2.  The `sample`
+function will handle the re-initialization of the fields once the system is provided.
+"""
+mutable struct LinSysBlkRowRandCyclic <: LinSysBlkRowSampler
+    n_blocks::Int64
+    order::Union{Vector{Int64}, Nothing}
+    blocks::Union{Vector{Vector{Int64}}, Nothing}
+end
```

Comment:
For a `struct` like this, it is possible to leave out `mutable` so long as `n_blocks` does not change. Julia allows for pointers in a `struct` to change even if the `struct` is immutable. I do not know how this will cope with the `Nothing` keyword or if we should just initialize with an empty `Vector{Int64}()`.

See https://docs.julialang.org/en/v1/manual/types/#Mutable-Composite-Types

This is a v0.2 question, not for right now.

Reply:
ok